pkg/pipeline/templates/controller.go.tmpl

```diff
@@ -23,7 +23,15 @@ import (
 )
 
 // Setup adds a controller that reconciles {{ .CRD.Kind }} managed resources.
-func Setup(mgr ctrl.Manager, l logging.Logger, rl workqueue.RateLimiter, s terraform.SetupFn, ws *terraform.WorkspaceStore, concurrency int) error {
+func Setup(mgr ctrl.Manager, l logging.Logger, rl workqueue.RateLimiter, s terraform.SetupFn, ws *terraform.WorkspaceStore, concurrency int, enabledAPIs []string) error {
```

Comment:
I think we'll need a similar logic in `AddToScheme` as well since we can't register schemes of CRDs that are not installed.

Would it be possible to handle all this enable/disable logic in `cmd/provider/main.go`? i.e. maybe accumulate all `AddToScheme` and `Setup` function calls in a map instead of calling them in `zz_register.go` and `zz_setup.go`, then do the filtering in `cmd/provider/main.go` and only call the ones that are enabled? We could possibly contain the whole enable/disable APIs logic in a single place this way.

Since this will affect all providers, at a later iteration, we can think of having some logic in crossplane-runtime to handle all this in a streamlined way.